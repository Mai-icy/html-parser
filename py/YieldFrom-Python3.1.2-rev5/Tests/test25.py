#
#   Test catching an exception thrown into a
#   subgenerator and returning a value
#

def inner():
   try:
       yield 1
   except ValueError:
       print("inner caught ValueError") #pass
   return 2

def outer():
   v = yield from inner()
   print("inner returned %r to outer" % v)
   yield v

g = outer()
print(next(g))              # prints 1
print(g.throw(ValueError))  # prints 2
