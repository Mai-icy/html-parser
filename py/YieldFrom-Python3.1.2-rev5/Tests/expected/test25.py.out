1
inner caught ValueError
inner returned 2 to outer
2
